src/Database/Adapter/SQL.php

```diff
@@ -911,4 +918,38 @@ public function getMaxIndexLength(): int
     {
         return 768;
     }
+
+    /**
+     * @param Query $query
+     * @return string
+     * @throws Exception
+     */
+    abstract protected function getSQLCondition(Query $query): string;
+
+    /**
+     * @param array<Query> $queries
+     * @param string $separator
+     * @return string
+     * @throws Exception
+     */
+    public function getSQLConditions(array $queries = [], string $separator = 'and'): string
```

Comment:
Let's use capitals for SQL keywords

```suggestion
public function getSQLConditions(array $queries = [], string $separator = 'AND'): string
```